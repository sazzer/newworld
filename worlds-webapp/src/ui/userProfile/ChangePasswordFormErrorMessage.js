// @flow

import React from 'react';
import {Message} from 'semantic-ui-react';
import {Interpolate} from 'react-i18next';

/** The message keys for the errors we can support */
const problemKeys = {
    'tag:grahamcox.co.uk,2018,users/problems/invalid-password': 'userProfile.password.messages.error.invalidPassword',
    'passwordMismatch': 'userProfile.password.messages.error.passwordMismatch',
    'undefined': 'userProfile.password.messages.error.unexpected'
};

/** The flow type representing the props for the Change Password Form message */
type ChangePasswordFormErrorMessageProps = {
    errorCode?: string
};

/**
 * The message to display for the change password save result
 */
export default function ChangePasswordFormErrorMessage(props: ChangePasswordFormErrorMessageProps) {
    let problemCode = props.errorCode || 'undefined';

    if (!problemKeys.hasOwnProperty(problemCode)) {
        problemCode = 'undefined';
    }
    const errorKey = problemKeys[problemCode];

    return (
        <Message error>
            <Interpolate i18nKey={errorKey} />
        </Message>
    )
}
