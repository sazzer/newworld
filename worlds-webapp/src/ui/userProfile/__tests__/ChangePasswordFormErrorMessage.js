// @flow

import React from 'react';
import {shallow} from 'enzyme';
import ChangePasswordFormErrorMessage from '../ChangePasswordFormErrorMessage';

/** Create the component to test */
function setup({errorCode}: {errorCode?: string} = {}) {
    return {
        element: shallow(<ChangePasswordFormErrorMessage errorCode={errorCode} />)
    };
}

describe('ChangePasswordFormErrorMessage', () => {
    it('Renders correctly with no error code', () => {
        const { element } = setup();

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly with an unrecognized error code', () => {
        const { element } = setup({errorCode: 'unrecognized'});

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly with a special code of passwordMismatch', () => {
        const { element } = setup({errorCode: 'passwordMismatch'});

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly with an expected code', () => {
        const { element } = setup({errorCode: 'tag:grahamcox.co.uk,2018,users/problems/invalid-password'});

        expect(element).toMatchSnapshot();
    });
});
