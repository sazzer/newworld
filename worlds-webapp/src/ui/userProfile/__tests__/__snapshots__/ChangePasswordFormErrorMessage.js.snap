// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordFormErrorMessage Renders correctly with a special code of passwordMismatch 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.password.messages.error.passwordMismatch"
  />
</Message>
`;

exports[`ChangePasswordFormErrorMessage Renders correctly with an expected code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.password.messages.error.invalidPassword"
  />
</Message>
`;

exports[`ChangePasswordFormErrorMessage Renders correctly with an unrecognized error code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.password.messages.error.unexpected"
  />
</Message>
`;

exports[`ChangePasswordFormErrorMessage Renders correctly with no error code 1`] = `
<Message
  error={true}
>
  <Interpolate
    className=""
    i18nKey="userProfile.password.messages.error.unexpected"
  />
</Message>
`;
